practice/flutter/flutter/logic.py

```diff
@@ -0,0 +1,167 @@
+    """Takes a search text, searches through all posted Flutts for Flutts that contain that text, returns a list of
+    Flutts that match.
+
+    >>> models.Flutt(author='Fred', body='2', timestamp=datetime.datetime(2016, 9, 2, 10, 28, 34), author_id=0).save()
+    >>> get_matches_by_search_text(2)
+    <QuerySet [Flutt(author='Fred', body='2', timestamp=2016-09-02 10:28:34+00:00, author_id=0)]>
+    >>> get_matches_by_search_text(3)
+    Traceback (most recent call last):
+    ...
+    LookupError: No matches
+    """
+    matches = models.Flutt.objects.filter(body__contains=search_text)
+    if not matches:
+        raise LookupError('No matches')
+    else:
+        return matches
+
+
+def login_user(request, username, password):
+    """Logs the user in.
+
+    Returns a LookupError if the username does not exist.
+
+    Returns a ValueError if the password was incorrect.
+
+    >>> user = User.objects.create_user('john', 'lennon@thebeatles.com', 'johnpassword')
+    >>> c = client.Client()
+    >>> c.login(username='john', password='johnpassword')
+    True
+    >>> login_user('request', 'john', 'floorb')
+    Traceback (most recent call last):
+    ...
+    ValueError: Incorrect password
+    >>> login_user('request', 'fee', 'foo')
+    Traceback (most recent call last):
+    ...
+    LookupError: That user does not exist.
+    """
+    user = authenticate(username=username, password=password)
+    if user is not None:
+        login(request, user)
+    try:
+        User.objects.get(username=username)
+    except User.DoesNotExist:
+        raise LookupError('That user does not exist.')
+    else:
+        raise ValueError('Incorrect password')
+
+
+def get_username_display(username):
+    """Takes in the user object tied to the request.
+
+    If the user is logged in, returns a string which notifies user of what they are logged in as.
+
+    If the user is not logged in, returns a string which notifies user that they are not logged in.
+
+    >>> get_username_display('goofus')
+    'You are currently logged in as goofus'
+    >>> get_username_display('')
+    'You are currently not logged in. Please login to add a Flutt.'
+    """
+    if username != '':
+        return "You are currently logged in as " + username
+    else:
+        return "You are currently not logged in. Please login to add a Flutt."
+
+
+def get_user_id(username):
+    """Takes in a username, returns the user id for the user who has that username.
+
+    If there is no user with that username, raises a LookupError.
+    >>> user = User.objects.create_user('john', 'lennon@thebeatles.com', 'johnpassword')
+    >>> user.save()
+    >>> get_user_id('john')
+    1
+    >>> get_user_id('food')
+    Traceback (most recent call last):
+    ...
+    LookupError: That user does not exist
+
+    """
+    try:
+        user = User.objects.get(username=username)
+        return user.id
+    except User.DoesNotExist:
+        raise LookupError('That user does not exist')
```

Comment:
In general, don't re-wrap exceptions unless you're specifically adding required information. `User.DoesNotExist` obviously means the user couldn't be found and might contain some useful info that is not being kept upon throwing this new exception, e.g. user name?